src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -969,49 +969,57 @@ protected void GenerateFindFirstChar()
             }
 
             // Generate length check.  If the input isn't long enough to possibly match, fail quickly.
+            // It's rare for min required length to be 0, but even if it is, we still output the check
+            // in case the current position has been bumped beyond the end; we also use this return
```

Comment:
Can you clarify (maybe give an example) the bit about bumping beyond the end?

I get that this method may be called at any position in the text because of repeated matches.